astropy/io/fits/util.py

```diff
@@ -982,3 +984,74 @@ def _get_array_mmap(array):
         if isinstance(base.base, mmap.mmap):
             return base.base
         base = base.base
+
+
+def get_posargs(function):
+    """Helper function to retrieve the positional parameters from a
+    function
+
+    inspect.getargspec is deprecated in Python 3, while
+    inspect.signature does not exist in Python 2. This function caters
+    for both.
+
+    """
+    try:
+        posargs = list(inspect.signature(function).parameters.keys())
+    except AttributeError:
+        posargs = inspect.getargspec(function).args
+
+    return posargs
+
+
+def replaced_argument(oldkeyword, newkeyword):
```

Comment:
I agree that this is better suited to be placed in `astropy.utils.decorators`, not here. And indeed it looks a lot cleaner using a decorator like this! :+1: